ui/panels-plugin/src/components/OptionsEditorTabs/OptionsEditorTabs.tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Tab, TabProps, Tabs, TabsProps, Box } from '@mui/material';
+import { useState } from 'react';
+import { TabPanel } from './TabPanel';
+
+interface BaseTabConfig {
+  /**
+   * Content rendered when the tab is active.
+   */
+  content: React.ReactNode;
+}
+
+interface OtherTabConfig extends BaseTabConfig {
+  id: string;
+  label: TabProps['label'];
+}
+
+type CommonTabId = 'query' | 'settings';
+
+/**
+ * Common tabs that are frequently used in the options editor across multiple
+ * plugins. The label and display order of these tabs is not configurable to
+ * avoid user experience inconsistencies across plugins.
+ */
+type CommonTabs = { [property in CommonTabId]?: BaseTabConfig };
+
+/**
+ * Custom tabs specified for a given plugin. They are displayed after common
+ * tabs.
+ */
+type OtherTabs = {
+  other?: OtherTabConfig[];
+};
+
+export type OptionsEditorTabsProps = {
+  tabs: CommonTabs & OtherTabs;
+};
+
+// Configuration of the order and labeling for common tabs across plugins
+// to enforce a consistent UX.
+const TAB_CONFIG = [
+  { id: 'query', label: 'Query' },
+  { id: 'settings', label: 'Settings' },
+] as const;
+
+export const OptionsEditorTabs = ({ tabs }: OptionsEditorTabsProps) => {
+  const [activeTab, setActiveTab] = useState(0);
+
+  const handleChange: TabsProps['onChange'] = (_, newValue) => {
+    setActiveTab(newValue);
+  };
+
+  // Normalize the common tabs that are managed via constants in this file
+  // and custom tabs that bring their own config into a consistent shape for
+  // rendering.
+  const commonTabs = TAB_CONFIG.filter((tabConfig) => {
+    // Only include common tabs that are specified.
+    return !!tabs[tabConfig.id];
+  }).map((tabConfig) => {
+    return {
+      ...tabConfig,
+      ...tabs[tabConfig.id],
+    };
+  });
+  const otherTabs = tabs?.other || [];
+  const normalizedTabs = [...commonTabs, ...otherTabs];
+
+  // TODO: check on divider color
+  return (
+    <>
+      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
```

Comment:
I remember some mentioning that the border colors in `perses` didn't match our designs, but I wasn't sure what I should set this to in order to be more consistent. Any pointers for that?

Reply:
Not 100% sure, changing colors in the Perses theme is also an option (we need to build out this theme more), I think I remember Christine making a PR recently with a change that sounds similar

Reply:
Found it, this thread is what I was thinking of! https://github.com/perses/perses/pull/676/files#r1001173186

Reply:
Yeah i think we need to update the perses theme soon since the dark mode is not as nice and we probably need Mohit's inputs

Reply:
👍 For now, I adjusted it to match the theme color variable Christine used in her PR, so it's consistent with the query editor.